### PR TITLE
Fix: gateway_server_address containing scheme or port

### LINKED ISSUE #5198: If the Gateway Server Address contains a scheme or port, the UI thinks the Gateway is connected to a different cluster
#### Summary
If the Gateway Server Address option on a Gateway contains a scheme or port (e.g. `wss://stack.example.com:8887`), the UI thinks the Gateway is connected to a different cluster and therefore doesn't show any activity, even though it could actually be connected to the same cluster. 

#### Steps to Reproduce
1. Set up a gateway and get it connected via Basics Station
2. Change the Gateway Server address from `stack.example.com` to `wss://stack.example.com:8887`
3. The UI now says that the Gateway is in a different cluster, even though it isn't

#### Environment
TTS v3.17.2 in Docker on a Ubuntu host (on EC2).

#### How do you propose to implement this?
By simply wrapping gateway.gateway_server_address in getHostFromUrl() (similarly to what is already done for gsConfig.base_url), this solves the issue, as it extracts the hostname from the full address and ignores the scheme and port. 

#### Can you do this yourself and submit a Pull Request?
I have created a branch that fixes this. Let me know if it's acceptable and I will make a PR. https://github.com/jameseden1/lorawan-stack/tree/fix/5198-gateway-server-address

### PR DESCRIPTION
**Pull request by onsolutionjames**
#### Summary

Closes #5198 

#### Changes

 - Use getHostFromUrl() on the gateway_server_address, to strip any included scheme, port, etc

#### Testing

...

##### Regressions

...

#### Notes for Reviewers

...

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ For details about compatibility between different releases, see the **Commitment
   - Join Servers using Backend Interfaces 1.1 (protocol `BI1.1`) must be configured with a `sender-ns-id` containing the EUI of the Network Server.
 - Fix `time.Duration` flags in CLI.
 - Gateway Server will no longer leave permanent gateway connection stats data on the registry when crashing.
+- Fix incorrect handling of gateway_server_address in the console when it contains a scheme or port
```
Comment (kschiffer) [inline CHANGELOG.md]:
```suggestion
- Console determining gateways as "Other cluster" even though using the same host if server addresses not matching exactly (e.g. due to using different host or scheme)
```

**Comment by kschiffer** [comment]
@onsolutionjames do you plan on making the suggested change? Otherwise, I will close and recreate the PR.

**Comment by onsolutionjames** [comment]
Hi @kschiffer, I'm not sure what changes you're referring to? On the original issue you said it was good to go, see here: https://github.com/TheThingsNetwork/lorawan-stack/issues/5198#issuecomment-1057063701

Let me know if there are any changes I need to make though and I'll get on to them ASAP

**Review by kschiffer — CHANGES_REQUESTED** [review]

**Comment by kschiffer** [comment]
Hi @onsolutionjames, very sorry. I only see now that I hadn't actually submitted my review :facepalm: .

Can you make the change to the changelog? Then we can merge.

**Comment by onsolutionjames** [comment]
Hi @kschiffer, I've just made this update to the changelog as requested. Let me know if there is anything else for me to do

**Review by kschiffer — APPROVED** [review]

**Comment by onsolutionjames** [comment]
Hi @kschiffer, it looks like this was added under the 3.18.0 heading in the changelog, which I'm guessing probably isn't ideal as 3.18.0 has already been released. Is there anything I should do to fix that? (or something I should have done to prevent it from happening?)

Screen Shot 2022-03-25 at 9 44 03 pm

**Comment by kschiffer** [comment]
Indeed. I'll fix that. Thank you!

**Pull request merged**